### PR TITLE
[WIP] .mailmap: update contributor list for squid

### PR DESCRIPTION
**Pull request by ljflores**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [x] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by ljflores** [comment]
Emails still need to be sent, but here are the changes.

**Comment by adk3798** [comment]
jenkins test make check

**Review by adk3798 — APPROVED** [review]

**Comment by ljflores** [comment]
> thanks @ljflores!
> 
> re: https://tracker.ceph.com/projects/ceph/wiki/Ceph_contributors_list_maintenance_guide#Fixing-the-contributor-list
> 
> > The following one-liner can be used to spot duplicates in the Commits by authors instead of looking for them manually
> 
> does this get rid of all the duplicate names in credits.sh output? does that one-liner return anything?
> 
> re: https://tracker.ceph.com/projects/ceph/wiki/Ceph_contributors_list_maintenance_guide#Sponsoring-organizations
> 
> have you verified that no individuals show up under the 'by organization' sections? if you see any names there, .organizationmap should map them to Unaffiliated (unless you can tell the real org)

I think I'm missing some changes.. more commits to come.

File: .mailmap [hunk 1]
```diff
@@ -19,6 +19,7 @@ Abhishek Lekshmanan <abhishek@suse.com> <alekshmanan@suse.com>
 Adam C. Emerson <aemerson@redhat.com>
 Adam King <adking@redhat.com> <kingamk3@gmail.com>
 Adam King <adking@redhat.com> <adking@redhat,com>
+Adam King <adking@redhat.com> <47704447+adk3798@users.noreply.github.com>
```
Comment (rzarzynski) [inline .mailmap:22]: The `noreply.github.com` items  look weird, to be honest. I looked over `git log` and they came from merges performed by the github's web interface. 

```
commit d9030299efd7ddfca81c1a3048d56b655fe7dc81
Merge: 14355eb39ad 187baffd56f
Author: Adam King <47704447+adk3798@users.noreply.github.com>
Date:   Wed Apr 3 13:06:55 2024 -0400

    Merge pull request #56628 from gbregman/main
    
    mgr/cephadm: add some NVMEof gateway fields to be configurable by cephadm
    
    Reviewed-by: Adam King <adking@redhat.com>
```

I don't have any strong feelings here, just was curious.

Comment (cbodley) [reply .mailmap:22]: the goal of mappings like this is to make sure that all of Adam's contributions get credited to a single individual. this change prevents entries like `Adam King <47704447+adk3798@users.noreply.github.com>` from showing up separately in the contributor credits

File: .mailmap [hunk 2]
```diff
@@ -102,6 +105,7 @@ Casey Bodley <cbodley@redhat.com> <cbodley@redat.com>
 Casey Bodley <cbodley@redhat.com> casey Bodley <cbodley@redhat.com>
 Casey Bodley <cbodley@redhat.com> Casey <cbodley@redhat.com>
 Casey Marshall <csm@soe.ucsc.edu> <rsdio@29311d96-e01e-0410-9327-a35deaab8ce9>
+Casey Bodley <cbodley@redhat.com> <cbodley@users.noreply.github.com>
```
Comment (idryomov) [inline .mailmap:108]: Should probably be moved up one line, so that all @cbodley's entries are together.

File: .mailmap [hunk 3]
```diff
@@ -562,11 +574,15 @@ Peng Cheng Zhang <pengcheng.zhang@easystack.cn>
 Peng Cheng Zhang <pengcheng.zhang@easystack.cn> <pc@localhost.localdomain>
 Peng Lai <penglaiyxy@gmail.com>
 Peng Lai <penglaiyxy@gmail.com> <penglaiyxy>
+Pere Diaz Bou <30913090+pereman2@users.noreply.github.com>
```
Comment (idryomov) [inline .mailmap:577]: This email address seems to be already accounted for two lines below?

File: .mailmap [hunk 4]
```diff
@@ -666,12 +689,14 @@ Shengjing Zhu <zhsj@umcloud.com> <i@zhsj.me>
...
 Shen, Hang <shenhang@kuaishou.com> shenhang <shenhang@kuaishou.com>
 Shilpa Jagannath <smanjara@redhat.com> Shilpa Manjarabad Jagannath <smanjara@smanjara.remote.csb>
+Shilpa Jagannath <smanjara@redhat.com> <smanjara@users.noreply.github.com>
 Shinobu Kinjo <shinobu@redhat.com> <ceph@ceph-stack.redhat.com>
 Shinobu Kinjo <shinobu@redhat.com> <shinobu@linux.com>
 Shiqi <m13913886148@gmail.com>
 Shiqi <m13913886148@gmail.com> <1454927420@qq.com>
 Shishir Gowda <shishir.gowda@sandisk.com>
 Shotaro Kawaguchi <kawaguchi.s@jp.fujitsu.com>
+shraddhaghatol <shraddha.j.ghatol@seagate.com> <root@ssc-vm-rhev4-2955.colo.seagate.com>
```
Comment (idryomov) [inline .mailmap:699]: Shraddha Ghatol?

File: .organizationmap [hunk 5]
```diff
@@ -360,6 +360,7 @@ IBM <contact@IBM.com> Samuel Matzek <smatzek@us.ibm.com>
 IBM <contact@IBM.com> Sunil Angadi <Sunil.Angadi@ibm.com>
 IBM <contact@IBM.com> Teoman Onay <tonay@ibm.com>
 IBM <contact@ibm.com> Ulrich Weigand <ulrich.weigand@de.ibm.com>
+IBM <contact@ibm.com> Shilpa Jagannath <smanjara@redhat.com>
```
Comment (idryomov) [inline .organizationmap:363]: There is another entry mapping to Red Hat below:
```
Red Hat <contact@redhat.com> Shilpa Jagannath <smanjara@redhat.com>
```

File: .peoplemap [hunk 6]
```diff
@@ -41,37 +47,82 @@ Ken Dreyer <kdreyer@redhat.com> Ken Dreyer <ken.dreyer@inktank.com>
...
 Ross Turk <rturk@redhat.com> Ross Turk <ross.turk@inktank.com>
+Sachin Punadikar <sachin.punadikar@ibm.com> <sachin.punadikar@seagate.com>
 Sage Weil <sweil@redhat.com> Sage Weil <sage@inktank.com>
-Sebastian Wagner <sewagner@redhat.com> Sebastian Wagner <swagner@suse.com>
+Sebastian Wagner <swagner@redhat.com> Sebastian Wagner <swagner@suse.com>
 Sahid Orentino Ferdjaoui <sahid.ferdjaoui@redhat.com> Sahid Orentino Ferdjaoui <sahid.ferdjaoui@cloudwatt.com>
 Samuel Just <sjust@redhat.com> Samuel Just <sam.just@inktank.com>
 Sandon Van Ness <svanness@redhat.com> Sandon Van Ness <sandon@inktank.com>
+Shang Ding <dingshang2013@163.com> <dingshang@xsky.com>
+Shilpa Jagannath <smanjara@redhat.com> <smanjara@li-b46676cc-2f05-11b2-a85c-9b92c8acacdf.ibm.com>
```
Comment (cbodley) [inline .peoplemap:87]: Shilpa's ibm addr is smanjara@ibm.com. you can use .mailmap to map this strange ibm addr to that, then .peoplemap can use \<smanjara@redhat.com\> \<smanjara@ibm.com\> and .organizationmap can map \<smanjara@ibm.com\> to IBM

cc @smanjara

**Comment by ljflores** [comment]
Added WIP to the title since I'm still adding more commits.

File: .mailmap (hunk 3 above)
Comment (ljflores) [reply .mailmap:614]: ACK

File: .mailmap (hunk 4 above)
Comment (ljflores) [reply .mailmap:739]: ACK

**Pull request closed without merging**